### PR TITLE
Golang runner task. Task to run go application

### PR DESCRIPTION
**Pull request by hililh**
Golang runner task. Task to run go application

### CODE REVIEW

File: Tasks/Go/task.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "id": "34B37FDD-BBF7-4EF1-B37C-9652CA7BB355",
+    "name": "Go",
+    "friendlyName": "Go",
+    "description": "Get, build, test a go application, or run a custom go command.",
+    "helpMarkDown": "[More Information](https://go.microsoft.com/fwlink/?linkid=867582)",
+    "category": "Package",
```
Comment (sachinma) [inline Tasks/Go/task.json]: The category should be build

File: Tasks/Go/task.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    "author": "Microsoft Corporation",
+    "preview": true,
+    "version": {
+        "Major": 0,
+        "Minor": 1,
+        "Patch": 0
+    },
+    "runsOn": [
+        "Agent",
+        "DeploymentGroup"
```
Comment (sachinma) [inline Tasks/Go/task.json]: Lets keep it consistent with .Net Core and have it only for agent. Can you confirm from Alok

File: Tasks/Go/task.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    ],
+    "demands": [
```
Comment (sachinma) [inline Tasks/Go/task.json]: should we add a demand?

File: Tasks/Go/package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "url": "git+https://github.com/microsoft/vsts-tasks.git"
+  },
+  "author": "Microsoft Corporation",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/microsoft/vsts-tasks/issues"
+  },
+  "homepage": "https://github.com/microsoft/vsts-tasks#readme",
+  "dependencies": {
+    "vso-node-api": "^6.2.2-preview",
```
Comment (sachinma) [inline Tasks/Go/package.json]: dont use it

File: Tasks/Go/main.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        this.argument = tl.getInput("arguments", false);
+        this.workingDir = tl.getInput("workingDirectory", false);
+        if (this.workingDir == null) {
+            this.workingDir = tl.getVariable("System.DefaultWorkingDirectory");
+        }
+        this.failOnStdErr = tl.getBoolInput("failOnStandardError", false);
+    }
+
+    public async execute() {
+        return new Promise<string>(async (resolve, reject) => {
```
Comment (sachinma) [inline Tasks/Go/main.ts]: You dont need this Promise wrapper.

File: Tasks/Go/main.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        this.command = tl.getInput("command", true).trim();
+        if (this.command == "custom") {
+            this.command = tl.getInput("customCommand", true).trim();
+        }
+        this.argument = tl.getInput("arguments", false);
+        this.workingDir = tl.getInput("workingDirectory", false);
+        if (this.workingDir == null) {
+            this.workingDir = tl.getVariable("System.DefaultWorkingDirectory");
+        }
+        this.failOnStdErr = tl.getBoolInput("failOnStandardError", false);
```
Comment (sachinma) [inline Tasks/Go/main.ts]: You dont have this option in task.json

File: Tasks/Go/main.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    constructor() {
+        this.command = tl.getInput("command", true).trim();
+        if (this.command == "custom") {
+            this.command = tl.getInput("customCommand", true).trim();
+        }
+        this.argument = tl.getInput("arguments", false);
+        this.workingDir = tl.getInput("workingDirectory", false);
+        if (this.workingDir == null) {
+            this.workingDir = tl.getVariable("System.DefaultWorkingDirectory");
```
Comment (sachinma) [inline Tasks/Go/main.ts]: I think system.defaultworkingdir is default in tool runner if workingDir is null

File: Tasks/Go/task.json (hunk 3 above)
Comment (hililh) [reply Tasks/Go/task.json]: Will add Go in agent capabilities later

**Comment by sachinma** [comment]
Add L0s

**Pull request merged**